File: Study/BAEKJOON/Basic Math2/Q9020.py
def isPrime(n):
    a = [True] * n
    for i in range(2, int(n ** 0.5) + 1):
        if a[i] == True:
            for j in range(i + i, n, i):
                a[j] = False
    return [i for i in range(2, n) if a[i] == True]

def prime(n):
    b = isPrime(n)
    c = max([i for i in range(len(b)) if b[i] <= n / 2])
    for i in range(c, -1, -1):
        for j in range(i, len(b)):
            if b[i] + b[j] == n:
                return [b[i], b[j]]

for i in range(int(input())):
    n = int(input())
    print(" ".join(map(str, prime(n))))

# 몰라서 보고했음

File: Study/BAEKJOON/Basic Math2/Q9020.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Q9020 {
    public static boolean[] prime = new boolean[10001];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringBuilder sb = new StringBuilder();

        isPrime();

        int T = Integer.parseInt(br.readLine());
        int start;
        int end;
        int n;

        for (int i = 0; i < T; i++) {
            n = Integer.parseInt(br.readLine());
            start = n / 2;
            end = n / 2;

            while (true) {
                if (!prime[start] && !prime[end]) {
                    sb.append(start).append(' ').append(end).append('\n');
                    break;
                }

                start--;
                end++;
            }
        }

        System.out.print(sb);
    }

    public static void isPrime() {
        prime[0] = true;
        prime[1] = true;

        for (int i = 2; i <= Math.sqrt(prime.length); i++) {
            if (prime[i]) continue;

            for (int j = i * i; j < prime.length; j += i) {
                prime[j] = true;
            }
        }
    }
}